core/src/main/java/com/google/cloud/sql/CredentialFactory.java

```diff
@@ -26,4 +32,55 @@ public interface CredentialFactory {
   String CREDENTIAL_FACTORY_PROPERTY = "cloudSql.socketFactory.credentialFactory";
 
   HttpRequestInitializer create();
+
+  /**
+   * Returns a GoogleCredentials to use for generating tokens and authenticating web requests. The
+   * default implementation will attempt to extract the GoogleCredentials object from known
+   * credential HttpRequestInitializer implementations already in used to provide tokens for IAM
+   * Authentication.
+   */
+  default GoogleCredentials getCredentials() {
```

Comment:
The default implementation of getCredentials() now has code that can extract a GoogleCredentials instance from well known implementations of HttpRequestInitializer. This code has been in use in `DefaultAccessTokenSupplier` for a long time to supply tokens for IAM Authentication.